### ISSUE TITLE
iOS 14 and some fonts 

### ISSUE THREAD

**Issue by c00lib1n** [google/fonts#2688]
Cyrrilic symbols in some fonts drawed by latin symbols after iOS14 update.
Only at iOS14, only certain fonts, only in direct html\css embedding (NOT with codepen or GoogleFonts fitting) 

For example:
https://fonts.google.com/specimen/Montserrat
https://fonts.google.com/specimen/Arsenal
https://fonts.google.com/specimen/Merriweather
https://fonts.google.com/specimen/Raleway

Test words: "Алая вспышка осветила силуэт зазубренного крыла."
'k' symbol must be 'к' 'ш' 'л' 'в' and another - taken from latin.

Left on picture - screenshot from iOS14 device, right - same page on DevTools

![93610016-2c038580-f9d5-11ea-8588-55b69157ecc2](https://user-images.githubusercontent.com/8288296/93739501-1a96c500-fbf1-11ea-8f8d-ef447b796f7e.png)

Also: 
https://github.com/JulietaUla/Montserrat/issues/109

**Comment by thlinard** [comment]
The example certainly doesn't show Latin characters, but the Bulgarian variant of Cyrillic characters. The behavior is reproducible with an test file using `lang="bg"`. If Safari (?) on iOS 14 does this by default, it's definitely a bug.

**Comment by c00lib1n** [comment]
Yes, it`s Safari iOS 14 and default embedding.

My test example:

`

  | <!DOCTYPE html>
-- | --
  | <html lang="ru">
  | <head>
  | <meta charset="windows-1251"/>
  | <title>head</title>
  |  
  |  
  |  
  | <link href="https://fonts.googleapis.com/css2?family=Alegreya+Sans&family=Arsenal&family=Exo+2&family=Merriweather&family=Montserrat&family=Noto+Sans&family=Open+Sans&family=Oswald&family=PT+Sans&family=Play&family=Raleway&family=Roboto&family=Roboto+Condensed&family=Rubik&family=Source+Sans+Pro&family=Spectral+SC:wght@200&display=swap" rel="stylesheet">
  |  
  |  
  | </head>
  | <body style="padding: 20px">
  |  
  |  
  |  
  | <div style="font-size: 32px; padding-bottom: 30px">
  | <b>Font By Default</b> <br>
  | Алая вспышка осветила силуэт зазубренного крыла.
  | </div>
  |  
  | <div style="font-family: sans-serif; font-size: 32px; padding-bottom: 30px">
  | <b>sans-serif By Default</b> <br>
  | Алая вспышка осветила силуэт зазубренного крыла.
  | </div>
  |  
  |  
  | <div style="font-family: 'Montserrat', sans-serif; font-size: 32px; padding-bottom: 30px">
  | <b>Montserrat</b><br>
  | Алая вспышка осветила силуэт зазубренного крыла.
  | </div>
  |  
  | <div style="font-family: 'Raleway', sans-serif; font-size: 32px; padding-bottom: 30px">
  | <b>Raleway</b><br>
  | Алая вспышка осветила силуэт зазубренного крыла.
  | </div>
  |  
  |  
  | <div style="font-family: 'Merriweather', sans-serif; font-size: 32px; padding-bottom: 30px">
  | <b>Merriweather</b> <br>
  | Алая вспышка осветила силуэт зазубренного крыла.
  | </div>
  |  
  | <div style="font-family: 'Arsenal', sans-serif; font-size: 32px; padding-bottom: 30px">
  | <b>Arsenal</b> <br>
  | Алая вспышка осветила силуэт зазубренного крыла.
  | </div>
  |  
  |  
  |  
  | </body>
  | </html>

`

**Comment by thlinard** [comment]
`lang="ru"` shoud not display these glyphs. Definitely a bug.

**Comment by c00lib1n** [comment]
lang="ru" or lang="bg" don`t change anything in render.
It is bug reproducing only at mobile iOS 14 :(  , desktop browser work correctly.

If i dowload this font (woff2 format) and direct inmplement on page (without googlefont API) - all worked out in iOS 14 and looks good.

I think, GoogleFont return specific fonts set depending at devices, and this set somehow incorrect with Montserrat or Merriweather. 
But PT Sans, for example, work correctly

It is bug in iOS14 (invalid request or device type?)  and bug in GoogleFont (on desktop you return all font, to mob iOS14 - not all :) ?)  and bug in font developers (maybe, set dont consist some certain files of font?)

**Comment by thlinard** [comment]
Well, you could be right. Safari on iOS 14 is new, and perhaps Google servers incorrectly detect it and serve the wrong files.

@davelab6: perhaps a bug on Google side?

**Comment by Grawl** [comment]
URLs of files are not different on iOS and in Chrome on macOS

**Comment by bonzoSPb** [comment]
Any solution?

**Comment by c00lib1n** [comment]
I`m uploaded fonts  to local path and apply equal sets to an devices.  Direct connection still working incorrect

**Comment by bonzoSPb** [comment]
@davelab6 can you fix this?

**Comment by gigantz** [comment]
To fix this bug, add this into your global style file
```css
html {
  -webkit-locale: auto;
}
```

**Comment by vblgn** [comment]
@gigantz i try it, but it not working

**Comment by Grawl** [comment]
Good workaround!

**Comment by baikov** [comment]
> To fix this bug, add this into your global style file
> 
> ```css
> * {
>   -webkit-locale: auto;
>   white-space: normal;
> }
> ```
> 
> ps. tested only for Montserrat font

I only used the first rule and it works in my case. 
```css
* {
  -webkit-locale: auto;
}
```
The second sometimes breaks the layout.
Tested for Montserrat.

**Comment by gigantz** [comment]
with only `-webkit-locale: auto` I still had problems with some texts with `white-space: pre-wrap` style.
This value `white-space: normal` fixed for it for me.

**Comment by 0leg5ergeev** [comment]
Thanks for your investigation! 
After today iOS update to 14.2, the issue is relevant again. 
I have `* {  -webkit-locale: auto;  white-space: normal;}` string in css, at iOS 14 and 14.1 Montserratfont works fine with cyrillic, but now at 14.2 Mobile Safari pushing this css:
![image](https://user-images.githubusercontent.com/22079558/98389941-c98a4580-2065-11eb-9723-8770aee6168d.png)

Any ideas on how to fix it?

**Comment by sergeyboldetsky** [comment]
After updating to iOS 14.2 `* { -webkit-locale: auto; white-space: normal;}` does not respond, and the font has returned to a curved format like this: [https://github.com/google/fonts/issues/2688#issue-705345614](https://github.com/google/fonts/issues/2688#issue-705345614)

**Comment by Slavsquat** [comment]
Yeah, there doesn't seem to be a workaround this time. I'm surprised not that many people have actually raised this issue as of yet. Virtually any website using one of the affected fonts in Cyrillic is hugely affected because the bugged font is quite unpleasant to look at.

**Comment by thlinard** [comment]
> Yeah, there doesn't seem to be a workaround this time. I'm surprised not that many people have actually raised this issue as of yet. Virtually any website using one of the affected fonts in Cyrillic is hugely affected because the bugged font is quite unpleasant to look at.

Well, I guess the opinion of Bulgarian users differs...

@davelab6 @m4rc1e: I tested with self-hosted fonts. The bug is definitely on Google's side. Cinzel issue (https://github.com/NDISCOVER/Cinzel/issues/4) shows that the problem can also affect the appearance of Latin characters (with a Turkish variant for the i).

It seems that Safari for iOS (not Safari for macOS, at least not on macOS 10.15.7) has become `locl`-aware.

**Comment by arcmotion** [comment]
Here is my workaround (just a stopgap measure):
```
const isIOS = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream;

if (isIOS) {
    document.body.classList.add('is-ios');
}
```
 Then in my stylesheet I bind to html attribute "lang" value and change font-family when locale changes:
```scss
body.is-ios{
        :lang(ru){
            font-family: Roboto, sans-serif; //or any other font you like
        }
    }
```

**Comment by kpushkov** [comment]
If you are looking for a temporary solution, using the ttf format helped me.
```
@font-face {
    font-family: 'Montserrat';
    font-style: italic;
    font-weight: 700;
    font-display: swap;
    src: local('Montserrat Bold Italic'), local('Montserrat-BoldItalic'), url(https://fonts.gstatic.com/s/montserrat/v15/JTUPjIg1_i6t8kCHKm459WxZcgvD-w.ttf) format('truetype');
}
```

**Comment by thlinard** [comment]
> If you are looking for a temporary solution, using the ttf format helped me.
> 
> ```
> @font-face {
>     font-family: 'Montserrat';
>     font-style: italic;
>     font-weight: 700;
>     font-display: swap;
>     src: local('Montserrat Bold Italic'), local('Montserrat-BoldItalic'), url(https://fonts.gstatic.com/s/montserrat/v15/JTUPjIg1_i6t8kCHKm459WxZcgvD-w.ttf) format('truetype');
> }
> ```

No need of ttf. You can use the WOFF2 from the primary repositories:

- Montserrat: https://github.com/JulietaUla/Montserrat/releases/tag/v7.210
- Raleway: https://github.com/theleagueof/raleway/releases/tag/4.101
- Merriweather: https://github.com/SorkinType/Merriweather/tree/master/fonts/woff2

**Comment by Slavsquat** [comment]
Look like this has been fixed. I don't see this behaviour on Google Fonts anymore, for any of the fonts mentioned in the original post.

**Comment by Slavsquat** [comment]
Google Fonts website gives me both variations (normal and broken) of Montserrat font depending on if there were originally Latin or Cyrillic symbols in the "Type something" field. 🤔

**Comment by sermelipharo** [comment]
Got into same issue on Safari _Version 14.0.1 (16610.2.11.51.8)_ with Big Sur _11.0.1 (20B29)_
![IMAGE 2020-11-16 10:19:34](https://user-images.githubusercontent.com/9950381/99223259-3a172c00-27f5-11eb-8845-74713d1bf051.jpg)
![IMAGE 2020-11-16 10:19:46](https://user-images.githubusercontent.com/9950381/99223268-40a5a380-27f5-11eb-8a5c-d42b2a784e5f.jpg)

**Comment by c00lib1n** [comment]
Confirm previous notice - desktop Safari inherit this problem behavior
Also, after Big Sur update my _local fonts solution_ lost bulletproofing - rouble symbol was replaced standard serif fonts  
<img width="332" alt="Screenshot 2020-11-16 at 21 20 54" src="https://user-images.githubusercontent.com/8288296/99292463-4da4b000-2852-11eb-87ee-75adde475d90.png">

**Comment by Crow26ru** [comment]
I think this problem is due to the wrong unicode-range for Cyrillic. I checked this bug with Cyrillic unicode-range from Google use Cyrillic subset for Montserrat font on iPhone 7. To fix this bug, download Montserrat font without subsets.

**Comment by thlinard** [comment]
> Got into same issue on Safari _Version 14.0.1 (16610.2.11.51.8)_ with Big Sur _11.0.1 (20B29)_

Hi @sermelipharo ,

Could you post a screenshot of the following page with Safari on macOS 11:
http://www.georgduffner.at/ebgaramond/tests/locltest.html

Thanks!

**Comment by sermelipharo** [comment]
looks fine to me
<img width="1536" alt="Screenshot 2020-11-18 at 13 10 52" src="https://user-images.githubusercontent.com/9950381/99516932-d2511480-299f-11eb-9fe0-a8cb5fe1f64c.png">

**Comment by thlinard** [comment]
> looks fine to me

Indeed!

Here is the same test with Safari 14.0.1 on macOS 10.15.7:

<img width="1231" alt="Safari 14 0 1 on macOS 10 15 7" src="https://user-images.githubusercontent.com/13808824/99517470-b47bb380-298f-11eb-995d-2198ab299d86.png">

So, as I thought, Safari 14, on iOS and macOS 11 (but not on macOS 10.15) has become `locl`-aware (which Firefox and Chrome were for a long time already — a `locl`-aware browser can display different glyphs depending on the language declared in the HTML code). But something doesn't satisfy Safari in the special version of the fonts served by Google's servers, and it displays alternative glyphs regardless of the declared language (Bulgarian variant for a Russian page, Turkish variant for an English page, etc.).

**Comment by sermelipharo** [comment]
yeah, it feels exactly like you've described

**Comment by parfilov** [comment]
@thlinard do you mean that it's Apple's issue and we should jiggle them?

**Comment by Grawl** [comment]
I can see this issue for Safari `14.0.1 (16610.2.11.51.8)` on macOS Big Sur `11.0.1 (20B29)`

`-webkit-locale: auto;` workaround is not working here.

No issue for Chrome or Firefox.

**Comment by parfilov** [comment]
Ok. I tried to connect Montserrat locally:
- ttf
- woff
- eot
Works great one by one or together.

**Comment by parfilov** [comment]
> I can see this issue for Safari `14.0.1 (16610.2.11.51.8)` on macOS Big Sur `11.0.1 (20B29)`
> 
> `-webkit-locale: auto;` workaround is not working here.
> 
> Even more, removing `lang='ru'` from `<html>` not works.
> 
> No issue for Chrome or Firefox.

Same issue

**Comment by thlinard** [comment]
> @thlinard do you mean that it's Apple's issue and we should jiggle them?

Well, my best guess is that it's a mix.

- Safari 14 on iOS and macOS 11 manages to correctly handle a **complete** font with a `locl` feature and to react correctly to `lang` declarations in HTML code: it displays Bulgarian glyphs for Bulgarian pages, Turkish glyphs for Turkish pages, etc.
- Google serves a special version, slimmed down (for performance reasons), and Safari doesn't like it, which activates the alternative glyphs of the `locl` feature whatever the value declared in the `lang` parameter: Google could try to send to Safari 14 a version that suits it, and on the other hand, Apple could try to fix their `locl` handling, as Firefox and Chrome (which have been `locl`-aware for a long time) do fine with files served by Google.

**Comment by WoW-rotfl** [comment]
I've replaced g fonts by https://www.cdnfonts.com/montserrat.font - temporary solution until it's fixed. Not so good compressed but acceptable

**Comment by shbov** [comment]
> I've replaced g fonts by https://www.cdnfonts.com/montserrat.font - temporary solution until it's fixed. Not so good compressed but acceptable

It worked for me.

**Comment by karanevsky** [comment]
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/59500992/100555331-be64b700-32ab-11eb-808e-e34b10cf041a.png">

Not a Googles font. Same problem with Cyrillic.
https://type.today/ru/signal_mono

**Comment by Grawl** [comment]
proof to confirm:

![image](https://user-images.githubusercontent.com/846774/100593368-118a4880-3344-11eb-8d65-0f52e4b3e048.png)

Opera on the left, Safari TP on the right. macOS 11 Big Sur.

**Comment by iksent** [comment]
Fonts from https://google-webfonts-helper.herokuapp.com/ with manual import works OK.

**Comment by thlinard** [comment]
>  [https://google-webfonts-helper](https://google-webfonts-helper.herokuapp.com/)

It would be to flee from Charybdis and fall into Scylla. google-webfonts-helper is obsolete when it comes to Variable Fonts (issue #2602 for example, or https://github.com/NDISCOVER/Exo-2.0/issues/11).

**Comment by yuliayuyu** [comment]
I've connected Montserrat from Adobe Fonts. In the settings of the web project in the section "Font Display" you need to install SWAP. The font is now displayed correctly. 

<img width="550" alt="Снимок экрана 2020-12-02 в 15 42 20" src="https://user-images.githubusercontent.com/34818780/100884982-2d682880-34c3-11eb-9b86-a114d3b66f22.png">
The solution suitable for Tilda.cc

**Comment by sklazer** [comment]
Same issue there.

Only helped including 
`<link href="http://fonts.cdnfonts.com/css/montserrat" rel="stylesheet">`
at the <head> block. Other solutions wont work for me

(WordPress)

**Comment by setlab** [comment]
Same problem with **Montserrat Alternates**

This works for both:
```
<link href="http://fonts.cdnfonts.com/css/montserrat" rel="stylesheet">
<link href="http://fonts.cdnfonts.com/css/montserrat-alternates" rel="stylesheet">
```

**Comment by delibash** [comment]
Confirming same for https://fonts.google.com/specimen/Lora Italic.

**Comment by delibash** [comment]
> > Yeah, there doesn't seem to be a workaround this time. I'm surprised not that many people have actually raised this issue as of yet. Virtually any website using one of the affected fonts in Cyrillic is hugely affected because the bugged font is quite unpleasant to look at.
> 
> Well, I guess the opinion of Bulgarian users differs...
> 
> @davelab6 @m4rc1e: I tested with self-hosted fonts. The bug is definitely on Google's side (at least, the slimmed down version served by Google servers doesn't please Safari). Cinzel issue ([NDISCOVER/Cinzel#4](https://github.com/NDISCOVER/Cinzel/issues/4)) shows that the problem can also affect the appearance of Latin characters (with a Turkish variant for the i).
> 
> According to my tests, it seems that Safari for iOS (not Safari for macOS, at least not on macOS 10.15.7) has become `locl`-aware, like Firefox and Chrome.
> 
> Edit: [a locl feature test page](http://www.georgduffner.at/ebgaramond/tests/locltest.html).

How does the opinion of Bulgarian users resolve this issue for Russian users? Cause that's where the problem lies.

**Comment by thlinard** [comment]
> How does the opinion of Bulgarian users resolve this issue for Russian users? Cause that's where the problem lies.

I was pointing out that Cyrillic is not only used in Russia, and what sounds horrible to a Russian speaker may seem natural to a Bulgarian.

But as for the problem, we've made some progress in explaining it: see https://github.com/google/fonts/issues/2688#issuecomment-729585042 and https://github.com/google/fonts/issues/2688#issuecomment-732762754

**Comment by TheRuslan** [comment]
Same problem with Manrope font.

iOS:
![photo_2020-12-15_19-01-19](https://user-images.githubusercontent.com/141221/102224676-133d3a00-3f08-11eb-91ea-800b557df338.jpg)

Android:
![photo_2020-12-15_19-01-31](https://user-images.githubusercontent.com/141221/102224755-2b14be00-3f08-11eb-834c-f77715054344.jpg)

**Comment by delibash** [comment]
> 
> I was pointing out that Cyrillic is not only used in Russia, and what sounds horrible to a Russian reader may seem natural to a Bulgarian.
> 
> But as for the problem, we've made some progress in explaining it: see [#2688 (comment)](https://github.com/google/fonts/issues/2688#issuecomment-729585042) and [#2688 (comment)](https://github.com/google/fonts/issues/2688#issuecomment-732762754)

Thanks for the reference. But I fail to see your point. Since the discussion is specifically about Russian not rendering correctly.

**Comment by thlinard** [comment]
> Thanks for the reference. But I fail to see your point. Since the discussion is specifically about Russian not rendering correctly.

No, the problem should not be reduced to that. https://github.com/NDISCOVER/Cinzel/issues/4 is the exact same problem and it affects English pages rendered with Turkish characters.

**Comment by m4rc1e** [comment]
Thanks for all the info folks. I will dig into this issue to see what's happening.

Just a pointer but we absolutely love minimal test cases :-). I'll make one up now.

**Comment by m4rc1e** [comment]
Alright, I've just taken a look. Thanks to all the posts above, especially @thlinard's.

**Exec summary:**
All Cyrillic families which include an OpenType locl feature with a BGR langsys are broken. We can see that Roboto, PT Sans, Oswald are not broken since they don’t have a BGR langsys in their cyrl locl feature.

**So are the fonts at fault?**
No. This wasn't an issue until IOS 14 was launched. All other Browsers seem to manage it just fine. There seems to be nothing wrong with how we’ve the declared the BGR features (95% sure, stil investigating)


**Why hasn’t this happened to all Cyrillic families?**
All the families which have this issue have been made using the Glyphsapp font editor. This editor allows designers to automatically generate OpenType feature code. I reckon all the locl features for these families were automatically generated by this app. Again, I don’t think there is anything wrong with the autogenerated locl features, I just think IOS 14 isn’t handling them properly. 

I’ll dig deeper since this issue is also affecting some Turkish fonts. I’ll edit this reply soon.

**Comment by thlinard** [comment]
> All the families which have this issue have been made using the Glyphsapp font editor.

I'm not sure it's a good lead either. Self-hosted WOFF and WOFF2 files don't seem to have this problem, which seems to only appear for files served by the API.

I had recommended these sources:

- Montserrat: https://github.com/JulietaUla/Montserrat/releases/tag/v7.210
- Raleway: https://github.com/theleagueof/raleway/releases/tag/4.101
- Merriweather: https://github.com/SorkinType/Merriweather/tree/master/fonts/woff2

**Comment by m4rc1e** [comment]
> All the families which have this issue have been made using the Glyphsapp font editor.

> >I'm not sure it's a good lead either. Self-hosted WOFF and WOFF2 files don't seem to have this problem, which seems to only appear for files served by the API.

Sorry @thlinard, I deleted my reply. I'll do another round of tests and reedit.

**Comment by m4rc1e** [comment]
Alright, I've found what seems to be the root cause.

The web fonts that we serve have been run through pyftsubset which as the name suggests will subset fonts. When I inspect the GSUB tables of the Cyrillic subsetted fonts, the `cyrl` ScriptRecord doesn't include a `DefaultLangSys`.

```XML
   18       <ScriptRecord index="1">
   19         <ScriptTag value="cyrl"/> <--- start of cyrl script
   20         <Script>
   21           <!-- LangSysCount=2 -->
   22           <LangSysRecord index="0">
   23             <LangSysTag value="BGR "/> <--- Jumps straight to the Bulgarian
   24             <LangSys>
   25               <ReqFeatureIndex value="65535"/>
   26               <!-- FeatureCount=1 -->
   27               <FeatureIndex index="0" value="0"/>
   28             </LangSys>
   29           </LangSysRecord>
   30           <LangSysRecord index="1">
   31             <LangSysTag value="SRB "/>
   32             <LangSys>
   33               <ReqFeatureIndex value="65535"/>
   34               <!-- FeatureCount=1 -->
   35               <FeatureIndex index="0" value="1"/>
   36             </LangSys>
   37           </LangSysRecord>
   38         </Script>
```

If I then inspect the fonts in our download zip, they include a `DefaultLangSys` for the `cyrl` script.

```XML
  29         <ScriptTag value="cyrl"/>
  30         <Script>
  31           <DefaultLangSys> <--- Default langsys for cyrl declared
  32             <ReqFeatureIndex value="65535"/>
  33             <!-- FeatureCount=10 -->
  34             <FeatureIndex index="0" value="0"/>
  35             <FeatureIndex index="1" value="1"/>
...
  44           </DefaultLangSys>
  45           <!-- LangSysCount=4 -->
  46           <LangSysRecord index="0">
  47             <LangSysTag value="BGR "/>
```

If I modify the webfonts to include an empty `DefaultLangSys` in the cyrl `ScriptRecord`, Safari IOS 14 behaves correctly.

```XML
9582         <ScriptTag value="cyrl"/>
9583         <Script>
9584           <DefaultLangSys> <--- hand inserted
9585             <ReqFeatureIndex value="65535"/>
9586             <!-- FeatureCount=0 -->   
9587           </DefaultLangSys> <---
9588           <LangSysRecord index="0">           
9589             <LangSysTag value="BGR "/>
9590             <LangSys>
```


![image](https://user-images.githubusercontent.com/7525512/102347076-cb65f380-3f97-11eb-904b-9c8ca4e04d34.png)

*no subsetting: Fonts downloadable from Google Fonts*
*web subset: Web Fonts served from Google Fonts*
*hand subset: Running pyftsubset on the downloadable fonts using the same config as our backend*
*hand subset fix: Duplicate of hand subset that includes the hand inserted `DefaultLangSys`*


I've read the [MS spec](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#script-list-table-and-script-record) regarding these tables a few times but I cannot work out if all `ScriptRecords` need to declare a `DefaultLangSys`? If every ScriptRecord needs a DefaultLangSys, we'll need to fix fontTools. I suspect this isn't true and clients should instead use the DFLT `ScriptRecord` if they cannot find the desired `LangSys`. If the latter is true Safari needs to fix their shaping.

I've attached the testcase and fonts. I'll flag some folks at Apple/fonttools to so see what they think.

[ios14_test.zip](https://github.com/google/fonts/files/5702568/ios14_test.zip)

**Comment by m4rc1e** [comment]
The [MS spec](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#script-table-and-language-system-record) says that a Script's `defaultLangSysOffset` may be NULL. I assume this means that we don't need to include a `DefaultLangSys` for every script apart from the DFLT script?

If this is the case, it means Safari's shaping on IOS 14 is broken and we're actually providing complaint fonts.

@anthrotype is my assumption above correct? for context see my post above.

**Comment by anthrotype** [comment]
Yes it can be NULL, the subsetter is probably dropping it because it's empty.

**Comment by m4rc1e** [comment]
Cool. Thanks Cosimo. I'll see if I can get the relevant folks at Apple to see this.

What concerns me is that even if Apple does fix this issue, these particular versions of IOS Safari 14 will always be broken. I guess we could write a script to hotfix our fonts or modify pyftsubset but again we're modifying the fonts which are ok just to fix a software issue.

**Comment by m4rc1e** [comment]
Apparently, this issue has been solved in iOS 14.3. Can any test?

**Comment by thlinard** [comment]
> Apparently, this issue has been solved in iOS 14.3. Can anyone test?

Hi Marc. I tested https://fonts.google.com/specimen/Montserrat?preview.text=%D0%94%D0%9B%D0%B2%D0%B3%D0%B4%D0%B6%D0%B7%D0%B8%D0%BA%D0%BB%D0%BF%D1%82%D1%88%D1%8E&preview.text_type=custom with Safari on iOS 14.3, the result doesn't show Bulgarian variants. I don't have macOS 11.1 on hand to test, however.

**Comment by sermelipharo** [comment]
Fixed on macOS 11.1

**Comment by parfilov** [comment]
Seems like fixed on iOS & iPadOS 14.3
